lib/Common/Core/ConfigParser.cpp

```diff
@@ -306,13 +306,23 @@ void ConfigParser::ParseRegistryKey(HKEY hk, CmdLineArgsParser &parser)
 #endif // _WIN32
 }
 
-
-void ConfigParser::ParseConfig(HANDLE hmod, CmdLineArgsParser &parser)
+void ConfigParser::ParseConfig(HANDLE hmod, CmdLineArgsParser &parser, const char16* strCustomConfigFile)
 {
 #if defined(ENABLE_DEBUG_CONFIG_OPTIONS) && CONFIG_PARSE_CONFIG_FILE
-    Assert(!_hasReadConfig);
+    Assert(!_hasReadConfig || strCustomConfigFile != nullptr);
     _hasReadConfig = true;
 
+    const char16* configFileName = strCustomConfigFile;
+    const char16* configFileExt = _u(""); /* in the custom config case,
+                                             ext is expected to be passed
+                                             in as part of the filename */
```

Comment:
nit: with a slight rewording you probably could fit this comment on the same line without it being too long